packages/analytics/src/integrations/Omnitracking/Omnitracking.js

```diff
@@ -215,29 +217,61 @@ class Omnitracking extends Integration {
         );
       }
       case analyticsTrackTypes.TRACK: {
-        precalculatedParameters = this.getPrecalculatedParametersForEvent(data);
-
-        if (!this.validateTrackingRequisites()) {
-          logger.error(
-            `[Omnitracking] - Event ${data.event} could not be tracked since it had no unique view id.
-             A possible cause is trying to track an event before tracking a page view. 
-             Make sure you are tracking events after page views are tracked.
-             This event will not be tracked with Omnitracking`,
-          );
-
-          return;
+        const eventMapperFn = trackEventsMapper[data.event];
+
+        if (eventMapperFn) {
+          const mappedEvents = eventMapperFn(data);
+
+          if (isArray(mappedEvents)) {
+            await Promise.all(
```

Comment:
I think the `await` is not needed because this function is called by analytics which does not await the result, so it is safe to return the promise or just have a return statement after the Promise.all call.

Reply:
fixed, thanks :)